src/pages/register/consumer.tsx

```diff
@@ -0,0 +1,255 @@
+'use client';
+
+import { useState } from 'react';
+import Avatar from '@mui/material/Avatar';
+import CssBaseline from '@mui/material/CssBaseline';
+import TextField from '@mui/material/TextField';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import Link from '@mui/material/Link';
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import Typography from '@mui/material/Typography';
+import Container from '@mui/material/Container';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { object as yupObject, string as yupString, date as yupDate } from 'yup';
+import { Formik, Form } from 'formik';
+import LoadingButton from '@mui/lab/LoadingButton';
+import Save from '@mui/icons-material/Save';
+import { auth } from '@/utils/firebase.config';
+import { RecaptchaVerifier, signInWithPhoneNumber } from 'firebase/auth';
+import dayjs from 'dayjs';
+
+interface ISignInValues {
+  firstName: string,
+  lastName: string,
+  mobile: string,
+  birthDate: Date,
+}
+
+const defaultValues: ISignInValues = {
+  firstName: '',
+  lastName: '',
+  mobile: '',
+  birthDate: dayjs().toDate(),
+};
+
+const signinSchema = yupObject({
+  firstName: yupString()
+    .required('Please input a first name.'),
+  lastName: yupString()
+    .required('Please input a last name.'),
+  mobile: yupString()
+    .required('Please input your mobile number.'),
+  birthDate: yupDate()
+    .max(
+      dayjs(),
+      'Please input a date earlier than current date.',
+    )
+    .required('Please input your birth date.'),
+});
+
+function Copyright(props: any) {
+  return (
+    <Typography variant='body2' color='text.secondary' align='center' {...props}>
+      {'Copyright © '}
+      <Link color='inherit' href='https://mui.com/'>
+        Your Website
+      </Link>{' '}
+      {new Date().getFullYear()}
+      {'.'}
+    </Typography>
+  );
+}
+
+export default function SignIn() {
```

Comment:
This appears to be for registration, should we rename the function tto be more relevant to Registration than `SignIn`?